### PR TITLE
Fix bug when only the first match was highlighted

### PR DESCRIPTION
**Pull request by bogdan0083**
Now all matches are highlighted:

Works for queryRegex as well:

### CODE REVIEW

File: src/providers.ts [hunk 1]
```diff
@@ -222,19 +222,27 @@ Total Files: ${Object.keys(files).length}\n`;
...
-        state.line,
-        padding + regexMatch.index! + regexMatch[0].length
-      );
-      this._highlights[docUriString].push(
-        new DocumentHighlight(highlightRange, DocumentHighlightKind.Read)
-      );
+
+      // run through each match and update position to add 
+      // hightlight for all matches
+      let index: number = 0;
```
Comment (thieman) [inline src/providers.ts]: Nit: I think the explicit `number` here is extraneous?

**Review by thieman — COMMENTED** [review]
Looks great, could you please add a line to the changelog as well?

File: src/providers.ts (hunk 1 above)
Comment (bogdan0083) [reply src/providers.ts]: Done! 👍

**Comment by bogdan0083** [comment]
Done! Added a line to the changelog and removed the explicit type definition.

**Comment by thieman** [comment]
Thanks! I'll take a look at this and your other PRs this weekend.

**Comment by thieman** [comment]
Merging with a small tweak over in https://github.com/thieman/better-search/pull/28

**Pull request closed without merging**